### PR TITLE
add case property recipients

### PR DESCRIPTION
**Pull request by calellowitz**
##### SUMMARY

https://dimagi-dev.atlassian.net/browse/USH-225

This adds two new recipients for conditional alerts.
1) Specify mobile worker by username
2) Specify email by username

The first works for both SMS or Email alerts, the second only for email (for obvious reasons)

### CODE REVIEW

**Comment by orangejenny** [comment]
Would it make sense to add more [test coverage](https://github.com/dimagi/commcare-hq/blob/master/corehq/messaging/scheduling/tests/test_recipients.py)?

**Comment by calellowitz** [comment]
> Would it make sense to add more [test coverage](https://github.com/dimagi/commcare-hq/blob/master/corehq/messaging/scheduling/tests/test_recipients.py)?

yes

File: corehq/messaging/scheduling/forms.py [hunk 1]
```diff
@@ -2944,6 +2954,10 @@ def add_initial_recipients(self, recipients, initial):
         for recipient_type, recipient_id in recipients:
             if recipient_type == CaseScheduleInstanceMixin.RECIPIENT_TYPE_CUSTOM:
                 initial['custom_recipient'] = recipient_id
+            if recipient_type == CaseScheduleInstanceMixin.RECIPIENT_TYPE_CASE_PROPERTY_USER:
+                initial['username_case_property'] = recipient_id
+            if recipient_type == CaseScheduleInstanceMixin.RECIPIENT_TYPE_CASE_PROPERTY_USER:
```
Comment (orangejenny) [inline corehq/messaging/scheduling/forms.py:2959]: This should be `RECIPIENT_TYPE_CASE_PROPERTY_EMAIL`, yes?

**Comment by orangejenny** [comment]
The `isinstance` checks, assuming that a recipient is an email if it's a string, seem dubious. Would it be feasible to instead update `ScheduleInstance.recipient` to check for the new recipient types and return the mobile worker?

**Comment by orangejenny** [comment]
^ Nevermind the last comment, I had mixed up the two new recipient types. The type checking is still a code smell, but I don't have a better suggestion at the moment.

**Comment by calellowitz** [comment]
Yeah, I didn't love that either. I think the alternative is some type of lightweight class with a `get_email` method, that we could wrap the returned email in.

**Comment by orangejenny** [comment]
> I think the alternative is some type of lightweight class with a get_email method

Yeah. I do like the idea of a little email class, but not certain it's worth the effort. One other thing such a class could do is validate the given email. What happens now if the email is invalid? I'm guessing it fails at the point where the email is sent, but I don't know if that error would end up in Message History since I think the email is sent asynchronously.

**Review by orangejenny — APPROVED** [review]

**Comment by orangejenny** [comment]
> What happens now if the email is invalid?

Offlined. Invalid emails with this recipient are treated the same as invalid emails from other recipients, such as if a mobile worker's email address was entered incorrectly.

**Review by orangejenny — APPROVED** [review]

**Pull request merged**